CONTRIBUTING.md

```diff
@@ -16,15 +16,186 @@ GitBook (which we use for documentation) uses Markdown as its file format.
 * Always use inline links.
 * Do not hard-wrap prose text.  Set your text editor to soft wrapping.
 * Internal links should be absolute (starting with `/`) and link to a file ending in `.md`, not `.html`. That makes them easier to find when needed to update.
+* Each page must contain a front matter section, whether it is a single or list (`_index.md`) file. See the **Front matter** section below for more details
+
+### Front matter
+
+In general, this project uses two types of files: "single" and "list". List pages are made using `_index.md` files in every subdirectory, whereas single pages are generated from every other page without that name.
+
+Additionally, single and list pages are built differently depending on their `layout`, which is defined in `layouts/_default/(single|list).html`. For our documentation, list pages by default generate buttons for each page in that subdirectory, whereas single pages are built from the contents of that file and include a table of contents.
+
+
+#### Adding pages
+
+If you wanted to add a new page for a runtime to the **Languages** section, you would add the following file:
+
+```text
+src/
+   ...
+   languages/
+      ...
+      new_language.md
+```
+
+`new_language.md` should contain at least the following in its front matter at the top of the page.
+
+```
+---
+title: "New language"
+weight: 4
+---
+```
+
+`title` sets the `<h1>` header for the page title, and weight defines in what order items in this subsection should be listed in the sidebar. Here the new page will appear fourth under the **Languages** section.
```

Comment:
To clarify: Is it higher number before lower number, or lower before high?  Make that explicit, because different systems do it differently, just to piss you off.

Reply:
lower number = higher in the sidebar. I'll  include